File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
     public int maxArea(int[] heights) {
        int left = 0, right = heights.length - 1;
        int a = 0, ma = 0;
        while (left < right) {
            a = (right - left) * Math.min(heights[left], heights[right]);
            if (ma < a) ma = a;
            if (heights[left] < heights[right]) left++;
            else right--;
        }
        return ma;
    }
}

File: Data Structures & Algorithms/trapping-rain-water/submission-0.cpp
class Solution {
public:
int trap(vector<int>& a) {
    if(a.size()==0)
        return 0;
	int n = a.size();
	int w = 0;
	vector<int>l(n, 0);
	vector<int>r(n, 0);
	l[0] = a[0];
	r[n - 1] = a[n - 1];
	for (int i = 1; i < n; i++)
		l[i] = max(l[i - 1], a[i]);
	for (int i = n - 2; i >= 0; i--)
		r[i] = max(r[i + 1], a[i]);

	for (int i = 0; i < n; i++)
		w += min(l[i], r[i]) - a[i];
	return w;
}
};